File: Python3/Medium/RuntimeError/valid-sudoku_7.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        
        # check rows
        for row in board:
            if not self.is_valid(row):
                return False
        # check cols
        for i in range(9):
            col = []
            for j in range(9):
                col.append(board[j][i])
            if not self.is_valid(col):
                return False
        # check 3x3
        for i in range(0, 9, 3):
            for j in range(0, 9, 3):
                box = []
                for k in range(i, i+3):
                    for l in range(j, j+3):
                        box.append(board[k][l])
                if not self.is_valid(box):
                    return False
        return True

File: Rust/Medium/Accepted/valid-sudoku_7.rs
impl Solution {
    pub fn is_valid_sudoku(board: Vec<Vec<char>>) -> bool {
        

        let mut rows = vec![vec![0; 9]; 9];
        let mut cols = vec![vec![0; 9]; 9];
        let mut blocks = vec![vec![0; 9]; 9];
        for i in 0..9 {
            for j in 0..9 {
                if board[i][j] == '.' {
                    continue;
                }
                let num = board[i][j].to_digit(10).unwrap() as usize - 1;
                let block_index = i / 3 * 3 + j / 3;
                rows[i][num] += 1;
                cols[j][num] += 1;
                blocks[block_index][num] += 1;
                if rows[i][num] > 1 || cols[j][num] > 1 || blocks[block_index][num] > 1 {
                    return false;
                }
            }
        }
        true
    }
}